lte/gateway/python/magma/pipelined/app/uplink_bridge.py

```diff
@@ -90,15 +90,26 @@ def initialize_on_connect(self, datapath):
 
         self._delete_all_flows()
         self._add_eth_port()
-        self._set_vlan_eth_port()
         self._setup_vlan_pop_dev()
         # flows to forward traffic between patch port to eth port
 
-        # 1. DHCP traffic
+        # 1.a. Setup SGi Vlan flows
+        if self.config.sgi_management_iface_vlan:
+            match = "in_port=%s,vlan_vid=%s/0x1fff" % (self.config.uplink_eth_port_name,
+                                                       hex(0x1000 | int(self.config.sgi_management_iface_vlan)))
+            actions = "strip_vlan,output:LOCAL"
+            self._install_flow(flows.MAXIMUM_PRIORITY, match, actions)
+
+            match = "in_port=LOCAL"
+            actions = "push_vlan:0x8100,mod_vlan_vid=%s,output:%s" % (self.config.sgi_management_iface_vlan,
+                                                                      self.config.uplink_eth_port_name)
+            self._install_flow(flows.MAXIMUM_PRIORITY, match, actions)
+
+        # 1.b. DHCP traffic
         match = "in_port=%s,ip,udp,tp_dst=68" % self.config.uplink_eth_port_name
```

Comment:
for my understanding what about dhcp port 67?

Reply:
This is about DHCP response. thats why it matches on 68.

Reply:
Do we have a different rule for discover?

Reply:
yes, discover is broadcasted msg. I was trying to reduce latency by having direct flow for DHCP return traffic. let me  explore better flows for DHCP req and response to keep it symmetric.